go/libraries/doltcore/row/noms_row.go

```diff
@@ -78,6 +80,12 @@ func pkRowFromNoms(sch schema.Schema, nomsKey, nomsVal types.Tuple) (Row, error)
 		if col.IsPartOfPK {
 			return false, errors.New("writing columns that are part of the primary key to non-pk values. col:" + col.Name)
 		} else if !types.IsNull(val) {
+			if err = validateSpatialTypeSRID(col, val); err != nil {
```

Comment:
It feels odd to be doing this spatial type validation at this level. This file is really all about noms io interface and feels like not the right place to do validation on specific types.